ddtrace/opentelemetry/span_test.go

```diff
@@ -14,71 +15,243 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/tinylib/msgp/msgp"
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
+	oteltrace "go.opentelemetry.io/otel/trace"
 
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/httpmem"
 )
 
-func TestSpanMethods(t *testing.T) {
-	testData := struct {
-		env, srv, oldOp, newOp string
-		tags                   [][]string // key - tags[0][0], value - tags[0][1]
-	}{
-		env:   "test_env",
-		srv:   "test_srv",
-		oldOp: "old_op",
-		newOp: "new_op",
-		tags:  [][]string{{"tag_1", "tag_1_val"}, {"opt_tag_1", "opt_tag_1_val"}}}
-	done := make(chan struct{})
-	var payload string
+type nullLogger struct{}
+
+func (n nullLogger) Log(_ string) {}
+
+func mockTracerProvider(t *testing.T, opts ...tracer.StartOption) (tp *TracerProvider, payloads chan string, cleanup func()) {
+	payloads = make(chan string)
 	s, c := httpmem.ServerAndClient(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch r.URL.Path {
 		case "/v0.4/traces":
+			if h := r.Header.Get("X-Datadog-Trace-Count"); h == "0" {
+				return
+			}
 			buf, err := io.ReadAll(r.Body)
-			if err != nil {
-				t.Fail()
+			if err != nil || len(buf) == 0 {
+				t.Fatalf("Test agent: Error receiving traces")
 			}
-			payload = fmt.Sprintf("%s", buf)
-			done <- struct{}{}
+			var js bytes.Buffer
+			msgp.UnmarshalAsJSON(&js, buf)
+			payloads <- js.String()
 		}
 		w.WriteHeader(200)
 	}))
-	defer s.Close()
+	opts = append(opts, tracer.WithHTTPClient(c), tracer.WithLogger(&nullLogger{}))
```

Comment:
The null logger doesn't really change the test workflow, not sure if it's necessary here.